test/jdk/com/sun/net/httpserver/bugs/TcpNoDelayNotRequired.java

```diff
@@ -0,0 +1,117 @@
+import jdk.test.lib.net.URIBuilder;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.*;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.logging.*;
+
+public class TcpNoDelayNotRequired {
+
+    public static final Logger LOGGER = Logger.getLogger("sun.net.www.protocol.http");
+
+    public static void main (String[] args) throws Exception {
+
+        java.util.logging.Handler outHandler = new StreamHandler(System.out,
+                new SimpleFormatter());
+        outHandler.setLevel(Level.FINEST);
+        LOGGER.setLevel(Level.FINEST);
+        LOGGER.addHandler(outHandler);
+
+        InetAddress loopback = InetAddress.getLoopbackAddress();
+        InetSocketAddress addr = new InetSocketAddress (loopback, 0);
+        HttpServer server = HttpServer.create (addr, 0);
+        HttpHandler handler = new Handler();
+        HttpContext ctx = server.createContext ("/test", handler);
+        HttpContext ctx2 = server.createContext ("/chunked", handler);
+        ExecutorService executor = Executors.newCachedThreadPool();
+        server.setExecutor (executor);
+        server.start ();
+
+        HttpClient client = HttpClient.newBuilder().build();
+
+        long start = System.currentTimeMillis();
+        for(int i=0;i<1000;i++) {
+            var uri = URIBuilder.newBuilder().scheme("http").loopback().port(server.getAddress().getPort()).path("/test").build();
+            var response = client.send(HttpRequest.newBuilder(uri).build(), HttpResponse.BodyHandlers.ofString());
+            if(!response.body().equals("hello")) throw new IllegalStateException("incorrect body");
+        }
+        for(int i=0;i<1000;i++) {
+            var uri = URIBuilder.newBuilder().scheme("http").loopback().port(server.getAddress().getPort()).path("/chunked").build();
+            var response = client.send(HttpRequest.newBuilder(uri).build(), HttpResponse.BodyHandlers.ofString());
+            if(!response.body().equals("hello")) throw new IllegalStateException("incorrect body");
+        }
+        long time = System.currentTimeMillis()-start;
+        System.out.println("time "+time);
+        server.stop(0);
+        executor.shutdown();
+    }
+
+    static class Handler implements HttpHandler {
+        public void handle (HttpExchange t)
+                throws IOException
+        {
+            InputStream is = t.getRequestBody();
+            Headers map = t.getRequestHeaders();
+            Headers rmap = t.getResponseHeaders();
+            while (is.read () != -1) ;
+            is.close();
+            rmap.add("content-type","text/plain");
+            t.sendResponseHeaders(200,5);
+            t.getResponseBody().write("hello".getBytes(StandardCharsets.ISO_8859_1));
+            t.getResponseBody().close();
```

Comment:
I don't have a strong preference, but would you be open to updating this (and the ChunkedHandler) to:

```java
try (InputStream is = t.getRequestBody()) {
    is.readAllBytes();
}
Headers reqHeaders = t.getRequestHeaders();
Headers respHeaders = t.getResponseHeaders();
respHeaders.add("content-type", "text/plain");
t.sendResponseHeaders(200,5);
try (OutputStream os = t.getResponseBody()) {
    os.write("hello".getBytes(StandardCharsets.ISO_8859_1));
}

```
(I haven't compiled it or done any testing with the proposed snippet)

Reply:
Isn’t consistency in the code base more important. I like the change but it seems there should be a single PR that changes all of the test cases to this format.

Reply:
Personally, I think that is not the correct choice. I think consistency is more important - especially in a code base this large. 

But I’ll do whatever you want. This is getting a bit frustrating. The PR review is 1% content and 99% personal nits. 

This is certainly a different approach than Google uses.

Reply:
To clarify the above. The standard should be - does this PR improve the code base. After which, if someone wishes to create a refactoring PR that addresses other issues - go for it. But don’t drag out someone’s already considerable effort in diagnosing and fixing the problem to satisfy other goals.

Clearly there is a lot of test code that could use cleanup - but this is not the way to go about doing that - holding up important fixes as an example. 

Most developers are going to look to existing code when implementing changes - if you have a “bunch of stuff in your head” you’d really like it to adhere to it is going to be a frustrating process and people won’t bother.